entities/Item.entity.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  ManyToOne,
+  JoinColumn,
+} from 'typeorm';
+import {
+  IsDefined,
+  IsUUID,
+  IsOptional,
+  Length,
+  IsDecimal,
+} from 'class-validator';
+import {
+  ApiHideProperty,
+  ApiProperty,
+  ApiPropertyOptional,
+} from '@nestjs/swagger';
+import { Auditable } from './Auditable';
+import { Company } from './Company.entity';
+
+@Entity('item')
+export class Item extends Auditable {
+  @PrimaryGeneratedColumn('uuid')
+  @IsOptional()
+  @IsUUID()
+  @ApiProperty({ description: 'Item Identifier' })
+  id?: string;
+
+  @Column()
+  @IsDefined()
+  @Length(2, 255)
+  @ApiProperty({ description: 'Item Name', minLength: 2, maxLength: 255 })
+  name: string;
+
+  @Column()
+  @IsDefined()
+  @IsDecimal()
+  @ApiPropertyOptional({
+    description: 'Item Cost',
+    type: IsDecimal,
+    nullable: false,
+  })
+  cost?: number;
```

Comment:
If you use the @IsDefined tag, why you define cost as optional?

Reply:
Done